pylint/checkers/refactoring.py

```diff
@@ -216,10 +222,39 @@ def visit_ifexp(self, _):
     def visit_comprehension(self, node):
         self._if_counter += len(node.ifs)
 
-    @utils.check_messages('too-many-nested-blocks', 'simplifiable-if-statement')
+    def _always_return(self, body):
+        """Search if a list of node has 'return', search nested return into ifs
+        """
+        alwr_all = []
+        for item in body:
+            if isinstance(item, astroid.Return):
+                return True
+            elif isinstance(item, astroid.If):
+                if not hasattr(item, 'alwr_if'):
+                    # Assign variable to object in order to re-use computed data
+                    item.alwr_if = self._always_return(item.body)
```

Comment:
This would be a bit simplified by using something as this:

```
@singledispatch
def _always_returns(node):
   scope = node.scope()
    for child in node.nodes_of_class(node):
         if isinstance(child, astroid.Return) and child.scope() is scope:
             return True
     return False

@_always_returns.register(astroid.If)
def _(node):
     body_returns = _always_returns(node.body) if ...
     orelse_returns = _always_returns(node.orelse) if ...
     return body_returns and orelse_returns
```